qcodes/instrument_drivers/yokogawa/GS200.py

```diff
@@ -10,29 +160,310 @@ class GS200(VisaInstrument):
+                           set_cmd=partial(self._get_set_output, "VOLT"),
+                           get_cmd=partial(self._get_set_output, "VOLT")
+                           )
+
+        self.add_parameter('current',
+                           label='Current',
+                           unit='I',
+                           set_cmd=partial(self._get_set_output, "CURR"),
+                           get_cmd=partial(self._get_set_output, "CURR")
+                           )
+
+        self.output_level = self.voltage   # This is changed through the source_mode interface
+
+        self.add_parameter('voltage_limit',
+                           label='Voltage Protection Limit',
+                           unit='V',
+                           vals=Ints(1, 30),
+                           get_cmd=":SOUR:PROT:VOLT?",
+                           set_cmd=":SOUR:PROT:VOLT {}",
+                           get_parser=float_round,
+                           set_parser=int)
+
+        self.add_parameter('current_limit',
+                           label='Current Protection Limit',
+                           unit='I',
+                           vals=Numbers(1e-3, 200e-3),
+                           get_cmd=":SOUR:PROT:CURR?",
+                           set_cmd=":SOUR:PROT:CURR {:.3f}",
                            get_parser=float,
-                           vals=Numbers(-10, 10))
+                           set_parser=float)
 
-        self.add_function('reset', call_cmd='*RST')
+        self.add_parameter('four_wire',
+                           label='Four Wire Sensing',
+                           get_cmd=':SENS:REM?',
+                           set_cmd=':SENS:REM {}',
+                           val_mapping={
+                              'off': 0,
+                              'on': 1,
+                           })
+        # Note: This feature can be used to remove common mode noise.
+        # Read the manual to see if you would like to use it
+        self.add_parameter('guard',
+                          label='Guard Terminal',
+                          get_cmd=':SENS:GUAR?',
+                          set_cmd=':SENS:GUAR {}',
+                          val_mapping={
+                              'off': 0,
+                              'on': 1,
+                          })
+
+        # Return measured line frequency
+        self.add_parameter("line_freq",
+                           label='Line Frequency',
+                           unit="Hz",
+                           get_cmd="SYST:LFR?",
+                           get_parser=int)
 
-        self.initialise()
+        # Check if monitor is present, and if so enable measurement
+        monitor_present = '/MON' in self.ask("*OPT?")
+        measure = GS200_Monitor(self, 'measure', monitor_present)
+        self.add_submodule('measure', measure)
+
+        # Reset function
+        self.add_function('reset', call_cmd='*RST')
```

Comment:
Why remove on/off functions here?

I personally find them convenient to use, but you prefer not to have them?

Reply:
The "add_function" method is rightly controversial as it does not add anything over just defining a function with "def function(self, ...)". What is the point of: 

self.add_function('on', call_cmd=self.on)
self.add_function('off', call_cmd=self.off)

Calling "on" will simply call the method "on", which it would do anyway even if we did not have the "add_function" calls. So why have them?